src/main/java/seedu/address/model/client/SuggestionType.java

```diff
@@ -44,6 +48,23 @@ public SuggestionType(String suggestionString) {
         }
     }
 
+    /**
+     * Returns this suggestion type's comparator.
+     */
+    public Comparator<Client> getSuggestionComparator() {
+        switch (suggestionString) {
+        case BY_AVAILABLE:
+            return (client1, client2) -> 0;
+        case BY_CONTRACT:
+            return Comparator.comparing(Client::getContractExpiryDate);
+        case BY_FREQUENCY:
+            return Comparator.comparing(Client::getLastModifiedInstant);
+        default:
+            assert false; // code execution will never reach here
+            return null;
+        }
+    }
+
     /**
      * Returns true if a given string is a valid tag name.
```

Comment:
Not part of your commit, but could you help to fix this typo in the comments

Reply:
done